Add 'Darwin' Support

When this library is built on Darwin/OS X/macos, the following error occurs:
```bash
$ go build .
# github.com/gdm85/go-libdeluge
./keep_alive.go:50:89: undefined: syscall.TCP_KEEPIDLE
./keep_alive.go:54:89: undefined: syscall.TCP_KEEPCNT
./keep_alive.go:58:89: undefined: syscall.TCP_KEEPINTVL
```

I believe this is because these constants in the `syscall` package are OS-dependent, and are not exposed on the Darwin platform.

--- a/keep_alive.go
+++ b/keep_alive.go
@@ -3,8 +3,6 @@ package delugeclient
 import (
 	"fmt"
 	"net"
-	"os"
-	"syscall"
 	"time"
 )
 
@@ -44,18 +42,6 @@ func enableKeepAlive(conn net.Conn, idleTime time.Duration, count int, interval 
 	}*/
 
 	return nil
-}
-
-func setIdle(fd int, secs int) error {
-	return os.NewSyscallError("setsockopt", syscall.SetsockoptInt(fd, syscall.IPPROTO_TCP, syscall.TCP_KEEPIDLE, secs))
-}
-
-func setCount(fd int, n int) error {
-	return os.NewSyscallError("setsockopt", syscall.SetsockoptInt(fd, syscall.IPPROTO_TCP, syscall.TCP_KEEPCNT, n))
-}
-
-func setInterval(fd int, secs int) error {
-	return os.NewSyscallError("setsockopt", syscall.SetsockoptInt(fd, syscall.IPPROTO_TCP, syscall.TCP_KEEPINTVL, secs))
 }
 
 func secs(d time.Duration) int {

--- a/keep_alive_darwin.go
+++ b/keep_alive_darwin.go
@@ -1,0 +1,24 @@
+package delugeclient
+
+import (
+	"os"
+	"syscall"
+)
+
+// from netinet/tcp.h (OS X 10.9.4)
+const (
+	_TCP_KEEPINTVL = 0x101 /* interval between keepalives */
+	_TCP_KEEPCNT   = 0x102 /* number of keepalives before close */
+)
+
+func setIdle(fd int, secs int) error {
+	return os.NewSyscallError("setsockopt", syscall.SetsockoptInt(fd, syscall.IPPROTO_TCP, syscall.TCP_KEEPALIVE, secs))
+}
+
+func setCount(fd int, n int) error {
+	return os.NewSyscallError("setsockopt", syscall.SetsockoptInt(fd, syscall.IPPROTO_TCP, _TCP_KEEPCNT, n))
+}
+
+func setInterval(fd int, secs int) error {
+	return os.NewSyscallError("setsockopt", syscall.SetsockoptInt(fd, syscall.IPPROTO_TCP, _TCP_KEEPINTVL, secs))
+}

--- a/keep_alive_linux.go
+++ b/keep_alive_linux.go
@@ -1,0 +1,18 @@
+package delugeclient
+
+import (
+	"os"
+	"syscall"
+)
+
+func setIdle(fd int, secs int) error {
+	return os.NewSyscallError("setsockopt", syscall.SetsockoptInt(fd, syscall.IPPROTO_TCP, syscall.TCP_KEEPIDLE, secs))
+}
+
+func setCount(fd int, n int) error {
+	return os.NewSyscallError("setsockopt", syscall.SetsockoptInt(fd, syscall.IPPROTO_TCP, syscall.TCP_KEEPCNT, n))
+}
+
+func setInterval(fd int, secs int) error {
+	return os.NewSyscallError("setsockopt", syscall.SetsockoptInt(fd, syscall.IPPROTO_TCP, syscall.TCP_KEEPINTVL, secs))
+}

